src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import core.basesyntax.shape.Shape;
+import core.basesyntax.supplier.ShapeSupplier;
+
+public class Main {
```

Comment:
For better separation of concerns, consider moving the shape generation and display logic to separate methods outside the main method.